src/pages/SignUp.tsx

```diff
@@ -1,318 +1,67 @@
-  { id: 'privacy' },
-  { id: 'actorInfo1' },
-  { id: 'actorInfo2' },
-  { id: 'offstageRoles' },
-  { id: 'profilePhoto' },
-  { id: 'demographics' },
-  { id: 'profilePreview' }
-];
-
-// flatten our step id's into a single array
-const flatSteps = (stepsArrObj: any) => stepsArrObj.map((step: any) => step.id);
-
-// establish our form data structure
-// assign defaults
-const defaultData = {
-  landingType: '',
-  stageRole: '',
-
-  basicsFirstName: '',
-  basicsLastName: '',
-  basicsEmailAddress: '',
-  basicsPassword: '',
-  basicsPasswordConfirm: '',
-  basics18Plus: false,
-
-  privacyAgreement: false,
-
-  actorInfo1Pronouns: '',
-  actorInfo1PronounsOther: '',
-  actorInfo1LGBTQ: '',
-  actorInfo1Ethnicities: [],
-
-  actorInfo2HeightFt: 0,
-  actorInfo2HeightIn: 0,
-  actorInfo2HeightNoAnswer: false,
-  actorInfo2AgeRanges: [],
-  actorInfo2Gender: '',
-  actorInfo2GenderRoles: [],
-  actorInfo2GenderTransition: '',
-
-  offstageRolesGeneral: [],
-  offstageRolesProduction: [],
-  offstageRolesScenicAndProperties: [],
-  offstageRolesLighting: [],
-  offstageRolesSound: [],
-  offstageRolesHairMakeupCostumes: [],
-
-  profilePhotoUrl: '',
-
-  demographicsUnionStatus: '',
-  demographicsUnionStatusOther: '',
-  demographicsAgency: '',
-  demographicsWebsites: [{ id: 1, url: '', websiteType: '' }], // { id, url, websiteType }
-  demographicsBio: ''
-};
-
-// default object to track a boolean true/false for which steps have form validation error states
-const createDefaultStepErrorsObj = (stepNames: any) => {
-  const stepErrorsObj: { [key: string]: boolean } = {};
-
-  // default all of our steps to false
-  // because the pages will update this themselves
-  // when errors or empty req fields arise or exist
-  stepNames.forEach((stepName: string) => {
-    stepErrorsObj[stepName] = false;
-  });
-
-  return stepErrorsObj;
-};
-
-const SignUp = () => {
-  const { firebaseAuth, firebaseFirestore } = useFirebaseContext();
-  const { setAccountRef, setProfileRef } = useProfileContext();
-  const [formData, setForm] = useForm(defaultData); // useForm is an extension of React hooks to manage form state
-  const [steps, setSteps] = useState(defaultSteps);
-  const [landingStep, setLandingStep] = useState(1); // Landing has two defaultSteps internally, based on if "individual"
-
-  // default state for form validation error states per step
-  const [stepErrors, setStepErrors] = useState(
-    createDefaultStepErrorsObj(flatSteps(steps))
+import AccountType from '../components/SignUp/AccountType';
+import GroupSignUp from '../components/SignUp/Group';
+import IndividualSignUp from '../components/SignUp/Individual';
+import {
+  ButtonCol,
+  PageFooterRow
+} from '../components/SignUp/SignUpFooterStyles';
+import Button from '../genericComponents/Button';
+
+const SignUp: React.FC = () => {
+  const [currentStep, setCurrentStep] = useState(-1);
+  const [accountType, setAccountType] = useState<'individual' | 'group' | null>(
```

Comment:
We might want to create some types files for defining these in one place

Reply:
Agreed.